modules/core/src/main/java/org/apache/synapse/api/rest/RestRequestHandler.java

```diff
@@ -60,7 +60,9 @@ public boolean process(MessageContext synCtx) {
         org.apache.axis2.context.MessageContext msgCtx = ((Axis2MessageContext) synCtx).
                 getAxis2MessageContext();
         String protocol = msgCtx.getIncomingTransportName();
-        if (!Constants.TRANSPORT_HTTP.equals(protocol) && !Constants.TRANSPORT_HTTPS.equals(protocol)) {
+        if (!Constants.TRANSPORT_HTTP.equals(protocol) && !Constants.TRANSPORT_HTTPS.equals(protocol)
+                && !BridgeConstants.TRANSPORT_HTTPWS.equals(protocol)
+                && !BridgeConstants.TRANSPORT_HTTPSWSS.equals(protocol)) {
```

Comment:
shall we have these 4 in a predefined list and check if it doesn't contain and do this..

Reply:
Fixed in [6aa6388](https://github.com/wso2/wso2-synapse/pull/1900/commits/6aa6388510737962df81431d8c85c1463e0ed4f5)